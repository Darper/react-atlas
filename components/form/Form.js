import React, { PropTypes } from 'react';

const propTypes = {
  children: PropTypes.node
};

const Form = ({ children, ...props }) => {
    return (
      <form {...this.props}>
        {children}
      </form>
    );
};

Form.propTypes = propTypes;

export default Form;
