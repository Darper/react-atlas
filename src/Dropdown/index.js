import { Dropdown } from "./Dropdown";
export { Dropdown };
import { Option } from "./Option";
export { Option };
