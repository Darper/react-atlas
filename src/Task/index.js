import { default as TaskCore } from "./Task";
export { TaskCore };
