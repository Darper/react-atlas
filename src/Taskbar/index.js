import { default as TaskbarCore } from "./Taskbar";
export { TaskbarCore };
