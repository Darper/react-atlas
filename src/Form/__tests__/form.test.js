import React from "react";
import { mount } from "enzyme";
import { FormCore } from "../../../react-atlas-core/src/Form/index";
import { TextFieldCore } from "../../../react-atlas-core/src/TextField";

describe("Test data", () => {
  it("Test submit data", function() {
    const onSubmit = jest.fn();
    const form = mount(
      <FormCore onSubmit={onSubmit}>
        <TextFieldCore name="UserField" value="Userdata" />
      </FormCore>
    );
    form.simulate("submit");

    expect(onSubmit).toBeCalled();

    expect(onSubmit.mock.calls[0][1]).toEqual({
      "UserField": "Userdata"
    });
  });

  it("Test submit data with multiple fields", function() {
    const onSubmit = jest.fn();
    const form = mount(
      <FormCore onSubmit={onSubmit}>
        <TextFieldCore name="UserField" value="Userdata" />
        <TextFieldCore name="UserField2" value="Userdata2" />
      </FormCore>
    );
    form.simulate("submit");

    expect(onSubmit).toBeCalled();

    expect(onSubmit.mock.calls[0][1]).toEqual({
      "UserField": "Userdata",
      "UserField2": "Userdata2"
    });
  });

  it("Test submit with nested fields", function() {
    const onSubmit = jest.fn();
    const form = mount(
      <FormCore onSubmit={onSubmit}>
        <TextFieldCore name="UserField" value="Userdata" />
        <div>
          <TextFieldCore name="NestedField" value="Userdata" />
        </div>
      </FormCore>
    );
    form.simulate("submit");

    expect(onSubmit).toBeCalled();

    expect(onSubmit.mock.calls[0][1]).toEqual(
      expect.objectContaining({
        "UserField": "Userdata",
        "NestedField": "Userdata"
      })
    );
  });

  it("Test submit with a nested field", function() {
    const onSubmit = jest.fn();
    const form = mount(
      <FormCore onSubmit={onSubmit}>
        <div>
          <TextFieldCore name="NestedField" value="Userdata" />
        </div>
      </FormCore>
    );
    form.simulate("submit");

    expect(onSubmit).toBeCalled();

    expect(onSubmit.mock.calls[0][1]).toEqual({
      "NestedField": "Userdata"
    });
  });

  it("Test that field onChange is called", function() {
    const onChange = jest.fn();
    const form = mount(
      <FormCore onSubmit={() => {}}>
        <div>
          <TextFieldCore
            name="NestedField"
            value="Userdata"
            onChange={onChange}
          />
        </div>
      </FormCore>
    );

    form
      .find("input")
      .at(0)
      .simulate("change", {
        "target": {
          "name": "NestedField",
          "value": "testdata"
        }
      });

    expect(onChange).toBeCalled();
  });

  it("Test onChangeHandler", function() {
    const onSubmit = jest.fn();
    const form = mount(
      <FormCore onSubmit={onSubmit}>
        <div>
          <TextFieldCore name="NestedField" value="Userdata" />
        </div>
      </FormCore>
    );
    form
      .find("input")
      .at(0)
      .simulate("change", {
        "target": {
          "name": "NestedField",
          "value": "testdata"
        }
      });
    form.simulate("submit");
    form.update();
    expect(onSubmit).toBeCalled();

    expect(onSubmit.mock.calls[0][1]).toEqual({
      "NestedField": "testdata"
    });
  });

  it("Test that onError is called when submitting an invalid field", function() {
    const onError = jest.fn();
    const form = mount(
      <FormCore onSubmit={() => {}} onError={onError}>
        <TextFieldCore name="UserField" value="Userdata" />
        <div>
          <TextFieldCore name="InvalidField" required value="" />
        </div>
      </FormCore>
    );
    form.simulate("submit");

    expect(onError).toBeCalled();
  });

  it("Test that required validation updates state properly", function() {
    const onError = jest.fn();
    const form = mount(
      <FormCore onSubmit={() => {}} onError={onError}>
        <TextFieldCore name="UserField" value="Userdata" />
        <div>
          <TextFieldCore name="InvalidField" required value="" />
        </div>
      </FormCore>
    );
    expect(form.state().childState.InvalidField.isValid).toBeTruthy();
    form.simulate("submit");

    expect(form.state().childState.InvalidField.isValid).toEqual(false);
  });

  // regression for https://github.com/DigitalRiver/react-atlas/issues/671
  it("Verify state is not lost on submission", function() {
    const form = mount(
      <FormCore>
        <TextFieldCore name="FieldOne" />
        <TextFieldCore name="FieldTwo" />
      </FormCore>
    );

    expect(form.state().childState.FieldOne.value).toBeUndefined();
    expect(form.state().childState.FieldTwo.value).toBeUndefined();

    form
      .find("input")
      .at(0)
      .simulate("change", {
        "target": {
          "name": "FieldOne",
          "value": "data one"
        }
      });

    form
      .find("input")
      .at(0)
      .simulate("change", {
        "target": {
          "name": "FieldTwo",
          "value": "data two"
        }
      });

    expect(form.state("childState").FieldOne.value).toEqual("data one");
    expect(form.state("childState").FieldTwo.value).toEqual("data two");
    expect(
      form
        .find("TextField")
        .at(0)
        .prop("name")
    ).toEqual("FieldOne");
    expect(
      form
        .find("TextField")
        .at(0)
        .prop("value")
    ).toEqual("data one");

    form.simulate("submit");
    const children = form.prop("children").concat(<TextFieldCore />);
    form.setProps({ children });

    expect(form.state().childState.FieldOne.value).toEqual("data one");
    expect(form.state().childState.FieldTwo.value).toEqual("data two");
  });
});
