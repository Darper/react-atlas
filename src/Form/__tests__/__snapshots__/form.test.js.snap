// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test correct render Test correct render 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Form
    method="POST"
    noValidate={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": WrappedComponent {
      "_reactInternalFiber": FiberNode {
        "_debugID": 5,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": FiberNode {
          "_debugID": 4,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": null,
          "alternate": null,
          "child": [Circular],
          "effectTag": 1,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {
              "method": "POST",
              "noValidate": true,
            },
          },
          "memoizedState": Object {
            "context": null,
            "mount": true,
            "props": Object {
              "method": "POST",
              "noValidate": true,
            },
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {
              "method": "POST",
              "noValidate": true,
            },
          },
          "ref": null,
          "return": FiberNode {
            "_debugID": 1,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": [Circular],
              "child": null,
              "effectTag": 0,
              "expirationTime": 1,
              "firstEffect": null,
              "index": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": null,
              "memoizedState": null,
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <form
                    class=""
                    method="POST"
                    novalidate=""
                  />
                </div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": Object {
                "baseState": null,
                "callbackList": null,
                "capturedValues": null,
                "expirationTime": 1,
                "first": Object {
                  "callback": null,
                  "capturedValue": null,
                  "expirationTime": 1,
                  "isForced": false,
                  "isReplace": false,
                  "next": null,
                  "partialState": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={
                        Object {
                          "method": "POST",
                          "noValidate": true,
                        }
                      }
                    />,
                  },
                },
                "hasForceUpdate": false,
                "isInitialized": false,
                "isProcessing": false,
                "last": Object {
                  "callback": null,
                  "capturedValue": null,
                  "expirationTime": 1,
                  "isForced": false,
                  "isReplace": false,
                  "next": null,
                  "partialState": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={
                        Object {
                          "method": "POST",
                          "noValidate": true,
                        }
                      }
                    />,
                  },
                },
              },
            },
            "child": [Circular],
            "effectTag": 32,
            "expirationTime": 0,
            "firstEffect": [Circular],
            "index": 0,
            "key": null,
            "lastEffect": [Circular],
            "memoizedProps": null,
            "memoizedState": Object {
              "element": <WrapperComponent
                Component={[Function]}
                context={null}
                props={
                  Object {
                    "method": "POST",
                    "noValidate": true,
                  }
                }
              />,
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "sibling": null,
            "stateNode": Object {
              "containerInfo": <div>
                <form
                  class=""
                  method="POST"
                  novalidate=""
                />
              </div>,
              "context": Object {},
              "current": [Circular],
              "finishedWork": null,
              "firstBatch": null,
              "hydrate": false,
              "nextScheduledRoot": null,
              "pendingChildren": null,
              "pendingCommitExpirationTime": 0,
              "pendingContext": null,
              "remainingExpirationTime": 0,
            },
            "tag": 3,
            "type": null,
            "updateQueue": Object {
              "baseState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "method": "POST",
                      "noValidate": true,
                    }
                  }
                />,
              },
              "callbackList": null,
              "capturedValues": null,
              "expirationTime": 0,
              "first": null,
              "hasForceUpdate": false,
              "isInitialized": true,
              "isProcessing": false,
              "last": null,
            },
          },
          "sibling": null,
          "stateNode": WrapperComponent {
            "_reactInternalFiber": [Circular],
            "_reactInternalInstance": Object {},
            "context": Object {},
            "props": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "refs": Object {},
            "state": Object {
              "context": null,
              "mount": true,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "updater": Object {
              "enqueueForceUpdate": [Function],
              "enqueueReplaceState": [Function],
              "enqueueSetState": [Function],
              "isMounted": [Function],
            },
          },
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "_debugSource": null,
        "alternate": null,
        "child": FiberNode {
          "_debugID": 6,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": [Circular],
          "_debugSource": undefined,
          "alternate": null,
          "child": null,
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "action": undefined,
            "children": undefined,
            "className": "",
            "encType": undefined,
            "method": "POST",
            "name": undefined,
            "noValidate": true,
            "onSubmit": [Function],
            "style": undefined,
            "target": undefined,
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "action": undefined,
            "children": undefined,
            "className": "",
            "encType": undefined,
            "method": "POST",
            "name": undefined,
            "noValidate": true,
            "onSubmit": [Function],
            "style": undefined,
            "target": undefined,
          },
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": <form
            class=""
            method="POST"
            novalidate=""
          />,
          "tag": 5,
          "type": "form",
          "updateQueue": null,
        },
        "effectTag": 1,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "method": "POST",
          "noValidate": true,
        },
        "memoizedState": Object {
          "childState": Object {},
        },
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "method": "POST",
          "noValidate": true,
        },
        "ref": null,
        "return": FiberNode {
          "_debugID": 4,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": null,
          "alternate": null,
          "child": [Circular],
          "effectTag": 1,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {
              "method": "POST",
              "noValidate": true,
            },
          },
          "memoizedState": Object {
            "context": null,
            "mount": true,
            "props": Object {
              "method": "POST",
              "noValidate": true,
            },
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {
              "method": "POST",
              "noValidate": true,
            },
          },
          "ref": null,
          "return": FiberNode {
            "_debugID": 1,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": [Circular],
              "child": null,
              "effectTag": 0,
              "expirationTime": 1,
              "firstEffect": null,
              "index": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": null,
              "memoizedState": null,
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <form
                    class=""
                    method="POST"
                    novalidate=""
                  />
                </div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": Object {
                "baseState": null,
                "callbackList": null,
                "capturedValues": null,
                "expirationTime": 1,
                "first": Object {
                  "callback": null,
                  "capturedValue": null,
                  "expirationTime": 1,
                  "isForced": false,
                  "isReplace": false,
                  "next": null,
                  "partialState": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={
                        Object {
                          "method": "POST",
                          "noValidate": true,
                        }
                      }
                    />,
                  },
                },
                "hasForceUpdate": false,
                "isInitialized": false,
                "isProcessing": false,
                "last": Object {
                  "callback": null,
                  "capturedValue": null,
                  "expirationTime": 1,
                  "isForced": false,
                  "isReplace": false,
                  "next": null,
                  "partialState": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={
                        Object {
                          "method": "POST",
                          "noValidate": true,
                        }
                      }
                    />,
                  },
                },
              },
            },
            "child": [Circular],
            "effectTag": 32,
            "expirationTime": 0,
            "firstEffect": [Circular],
            "index": 0,
            "key": null,
            "lastEffect": [Circular],
            "memoizedProps": null,
            "memoizedState": Object {
              "element": <WrapperComponent
                Component={[Function]}
                context={null}
                props={
                  Object {
                    "method": "POST",
                    "noValidate": true,
                  }
                }
              />,
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "sibling": null,
            "stateNode": Object {
              "containerInfo": <div>
                <form
                  class=""
                  method="POST"
                  novalidate=""
                />
              </div>,
              "context": Object {},
              "current": [Circular],
              "finishedWork": null,
              "firstBatch": null,
              "hydrate": false,
              "nextScheduledRoot": null,
              "pendingChildren": null,
              "pendingCommitExpirationTime": 0,
              "pendingContext": null,
              "remainingExpirationTime": 0,
            },
            "tag": 3,
            "type": null,
            "updateQueue": Object {
              "baseState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "method": "POST",
                      "noValidate": true,
                    }
                  }
                />,
              },
              "callbackList": null,
              "capturedValues": null,
              "expirationTime": 0,
              "first": null,
              "hasForceUpdate": false,
              "isInitialized": true,
              "isProcessing": false,
              "last": null,
            },
          },
          "sibling": null,
          "stateNode": WrapperComponent {
            "_reactInternalFiber": [Circular],
            "_reactInternalInstance": Object {},
            "context": Object {},
            "props": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "refs": Object {},
            "state": Object {
              "context": null,
              "mount": true,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "updater": Object {
              "enqueueForceUpdate": [Function],
              "enqueueReplaceState": [Function],
              "enqueueSetState": [Function],
              "isMounted": [Function],
            },
          },
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "sibling": null,
        "stateNode": [Circular],
        "tag": 2,
        "type": [Function],
        "updateQueue": null,
      },
      "_reactInternalInstance": Object {},
      "context": Object {},
      "onChangeHandler": [Function],
      "props": Object {
        "method": "POST",
        "noValidate": true,
      },
      "refs": Object {},
      "state": Object {
        "childState": Object {},
      },
      "submitHandler": [Function],
      "updater": Object {
        "enqueueForceUpdate": [Function],
        "enqueueReplaceState": [Function],
        "enqueueSetState": [Function],
        "isMounted": [Function],
      },
      "validate": [Function],
    },
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "method": "POST",
      "noValidate": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": <form
        class=""
        method="POST"
        novalidate=""
      />,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": undefined,
        "children": undefined,
        "className": "",
        "encType": undefined,
        "method": "POST",
        "name": undefined,
        "noValidate": true,
        "onSubmit": [Function],
        "style": undefined,
        "target": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": WrappedComponent {
        "_reactInternalFiber": FiberNode {
          "_debugID": 5,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": FiberNode {
            "_debugID": 4,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": null,
            "child": [Circular],
            "effectTag": 1,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "memoizedState": Object {
              "context": null,
              "mount": true,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "ref": null,
            "return": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": FiberNode {
                "_debugID": 1,
                "_debugIsCurrentlyTiming": false,
                "_debugOwner": null,
                "_debugSource": null,
                "alternate": [Circular],
                "child": null,
                "effectTag": 0,
                "expirationTime": 1,
                "firstEffect": null,
                "index": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": null,
                "memoizedState": null,
                "mode": 0,
                "nextEffect": null,
                "pendingProps": null,
                "ref": null,
                "return": null,
                "sibling": null,
                "stateNode": Object {
                  "containerInfo": <div>
                    <form
                      class=""
                      method="POST"
                      novalidate=""
                    />
                  </div>,
                  "context": Object {},
                  "current": [Circular],
                  "finishedWork": null,
                  "firstBatch": null,
                  "hydrate": false,
                  "nextScheduledRoot": null,
                  "pendingChildren": null,
                  "pendingCommitExpirationTime": 0,
                  "pendingContext": null,
                  "remainingExpirationTime": 0,
                },
                "tag": 3,
                "type": null,
                "updateQueue": Object {
                  "baseState": null,
                  "callbackList": null,
                  "capturedValues": null,
                  "expirationTime": 1,
                  "first": Object {
                    "callback": null,
                    "capturedValue": null,
                    "expirationTime": 1,
                    "isForced": false,
                    "isReplace": false,
                    "next": null,
                    "partialState": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={
                          Object {
                            "method": "POST",
                            "noValidate": true,
                          }
                        }
                      />,
                    },
                  },
                  "hasForceUpdate": false,
                  "isInitialized": false,
                  "isProcessing": false,
                  "last": Object {
                    "callback": null,
                    "capturedValue": null,
                    "expirationTime": 1,
                    "isForced": false,
                    "isReplace": false,
                    "next": null,
                    "partialState": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={
                          Object {
                            "method": "POST",
                            "noValidate": true,
                          }
                        }
                      />,
                    },
                  },
                },
              },
              "child": [Circular],
              "effectTag": 32,
              "expirationTime": 0,
              "firstEffect": [Circular],
              "index": 0,
              "key": null,
              "lastEffect": [Circular],
              "memoizedProps": null,
              "memoizedState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "method": "POST",
                      "noValidate": true,
                    }
                  }
                />,
              },
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <form
                    class=""
                    method="POST"
                    novalidate=""
                  />
                </div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": Object {
                "baseState": Object {
                  "element": <WrapperComponent
                    Component={[Function]}
                    context={null}
                    props={
                      Object {
                        "method": "POST",
                        "noValidate": true,
                      }
                    }
                  />,
                },
                "callbackList": null,
                "capturedValues": null,
                "expirationTime": 0,
                "first": null,
                "hasForceUpdate": false,
                "isInitialized": true,
                "isProcessing": false,
                "last": null,
              },
            },
            "sibling": null,
            "stateNode": WrapperComponent {
              "_reactInternalFiber": [Circular],
              "_reactInternalInstance": Object {},
              "context": Object {},
              "props": Object {
                "Component": [Function],
                "context": null,
                "props": Object {
                  "method": "POST",
                  "noValidate": true,
                },
              },
              "refs": Object {},
              "state": Object {
                "context": null,
                "mount": true,
                "props": Object {
                  "method": "POST",
                  "noValidate": true,
                },
              },
              "updater": Object {
                "enqueueForceUpdate": [Function],
                "enqueueReplaceState": [Function],
                "enqueueSetState": [Function],
                "isMounted": [Function],
              },
            },
            "tag": 2,
            "type": [Function],
            "updateQueue": null,
          },
          "_debugSource": null,
          "alternate": null,
          "child": FiberNode {
            "_debugID": 6,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": [Circular],
            "_debugSource": undefined,
            "alternate": null,
            "child": null,
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "action": undefined,
              "children": undefined,
              "className": "",
              "encType": undefined,
              "method": "POST",
              "name": undefined,
              "noValidate": true,
              "onSubmit": [Function],
              "style": undefined,
              "target": undefined,
            },
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {
              "action": undefined,
              "children": undefined,
              "className": "",
              "encType": undefined,
              "method": "POST",
              "name": undefined,
              "noValidate": true,
              "onSubmit": [Function],
              "style": undefined,
              "target": undefined,
            },
            "ref": null,
            "return": [Circular],
            "sibling": null,
            "stateNode": <form
              class=""
              method="POST"
              novalidate=""
            />,
            "tag": 5,
            "type": "form",
            "updateQueue": null,
          },
          "effectTag": 1,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "method": "POST",
            "noValidate": true,
          },
          "memoizedState": Object {
            "childState": Object {},
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "method": "POST",
            "noValidate": true,
          },
          "ref": null,
          "return": FiberNode {
            "_debugID": 4,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": null,
            "child": [Circular],
            "effectTag": 1,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "memoizedState": Object {
              "context": null,
              "mount": true,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "method": "POST",
                "noValidate": true,
              },
            },
            "ref": null,
            "return": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": FiberNode {
                "_debugID": 1,
                "_debugIsCurrentlyTiming": false,
                "_debugOwner": null,
                "_debugSource": null,
                "alternate": [Circular],
                "child": null,
                "effectTag": 0,
                "expirationTime": 1,
                "firstEffect": null,
                "index": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": null,
                "memoizedState": null,
                "mode": 0,
                "nextEffect": null,
                "pendingProps": null,
                "ref": null,
                "return": null,
                "sibling": null,
                "stateNode": Object {
                  "containerInfo": <div>
                    <form
                      class=""
                      method="POST"
                      novalidate=""
                    />
                  </div>,
                  "context": Object {},
                  "current": [Circular],
                  "finishedWork": null,
                  "firstBatch": null,
                  "hydrate": false,
                  "nextScheduledRoot": null,
                  "pendingChildren": null,
                  "pendingCommitExpirationTime": 0,
                  "pendingContext": null,
                  "remainingExpirationTime": 0,
                },
                "tag": 3,
                "type": null,
                "updateQueue": Object {
                  "baseState": null,
                  "callbackList": null,
                  "capturedValues": null,
                  "expirationTime": 1,
                  "first": Object {
                    "callback": null,
                    "capturedValue": null,
                    "expirationTime": 1,
                    "isForced": false,
                    "isReplace": false,
                    "next": null,
                    "partialState": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={
                          Object {
                            "method": "POST",
                            "noValidate": true,
                          }
                        }
                      />,
                    },
                  },
                  "hasForceUpdate": false,
                  "isInitialized": false,
                  "isProcessing": false,
                  "last": Object {
                    "callback": null,
                    "capturedValue": null,
                    "expirationTime": 1,
                    "isForced": false,
                    "isReplace": false,
                    "next": null,
                    "partialState": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={
                          Object {
                            "method": "POST",
                            "noValidate": true,
                          }
                        }
                      />,
                    },
                  },
                },
              },
              "child": [Circular],
              "effectTag": 32,
              "expirationTime": 0,
              "firstEffect": [Circular],
              "index": 0,
              "key": null,
              "lastEffect": [Circular],
              "memoizedProps": null,
              "memoizedState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "method": "POST",
                      "noValidate": true,
                    }
                  }
                />,
              },
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <form
                    class=""
                    method="POST"
                    novalidate=""
                  />
                </div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": Object {
                "baseState": Object {
                  "element": <WrapperComponent
                    Component={[Function]}
                    context={null}
                    props={
                      Object {
                        "method": "POST",
                        "noValidate": true,
                      }
                    }
                  />,
                },
                "callbackList": null,
                "capturedValues": null,
                "expirationTime": 0,
                "first": null,
                "hasForceUpdate": false,
                "isInitialized": true,
                "isProcessing": false,
                "last": null,
              },
            },
            "sibling": null,
            "stateNode": WrapperComponent {
              "_reactInternalFiber": [Circular],
              "_reactInternalInstance": Object {},
              "context": Object {},
              "props": Object {
                "Component": [Function],
                "context": null,
                "props": Object {
                  "method": "POST",
                  "noValidate": true,
                },
              },
              "refs": Object {},
              "state": Object {
                "context": null,
                "mount": true,
                "props": Object {
                  "method": "POST",
                  "noValidate": true,
                },
              },
              "updater": Object {
                "enqueueForceUpdate": [Function],
                "enqueueReplaceState": [Function],
                "enqueueSetState": [Function],
                "isMounted": [Function],
              },
            },
            "tag": 2,
            "type": [Function],
            "updateQueue": null,
          },
          "sibling": null,
          "stateNode": [Circular],
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "_reactInternalInstance": Object {},
        "context": Object {},
        "onChangeHandler": [Function],
        "props": Object {
          "method": "POST",
          "noValidate": true,
        },
        "refs": Object {},
        "state": Object {
          "childState": Object {},
        },
        "submitHandler": [Function],
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
        "validate": [Function],
      },
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "method": "POST",
        "noValidate": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": <form
          class=""
          method="POST"
          novalidate=""
        />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": undefined,
          "children": undefined,
          "className": "",
          "encType": undefined,
          "method": "POST",
          "name": undefined,
          "noValidate": true,
          "onSubmit": [Function],
          "style": undefined,
          "target": undefined,
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
